parcels/kernel.py

```diff
@@ -300,9 +318,6 @@ def execute_python(self, pset, endtime, dt):
                     if res is ErrorCode.Success and particles.state != state_prev:
                         res = particles.state
 
-                    if res == ErrorCode.Success and not np.isclose(particles.dt, pdt_prekernels):
-                        res = ErrorCode.Repeat
-
```

Comment:
hm, clearing this line kind'a disallows kernel repetition. The line's intent is actually that, if a particle is not advected by dt in full (e.g if the write-out timer to file or the plotting timer is between `particles.time` and `particles.time+particles.dt`), then the computation is continued until a full time step has been performed. I see how that clashes with the analytical advection, but clearing out this line regardless of analytical or iterative optimization looks like a potential error source.

Reply:
OK, I changed to 
```python
if not analytical and res == ErrorCode.Success and not np.isclose(particles.dt, pdt_prekernels):
    res = ErrorCode.Repeat
```